### PR TITLE
[#24] [Chore] Initial setup of project.

### LINKED ISSUE #24: [Chore] Setup the Project with required dependencies
## Why

Set up the project with dependencies such as Next JS, React, Typescript, testing frameworks

### PR DESCRIPTION
**Pull request by Prabeshpd**
- Close #24 

## What happened 👀

Setup the project with dependencies required for the project. In the dependency the basic setup for test has been added along with the `eslint` configs and `prettier` formatter. Basic directory or the project structure has also been added.

## Insight 📝

- Setup the project with next
- Add the basic dependencies such as typescript, react, react-dom and so on.
- Add the dependencies for test (Jest setup)
- Setup basic test for render page
- Setup basic project structure

## Proof Of Work 📹
Screen Shot 2566-07-20 at 10 28 31

### CODE REVIEW

File: public/next.svg
Comment (olivierobert) [inline public/next.svg:1]: Let's replace all placeholder images with Nimble's branding: https://drive.google.com/drive/folders/1NV8IRt5Bpxv73X-DNY37i3V1GAt0wCGd

File: package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (Prabeshpd) [inline package.json:1]: Add nimble eslint package

File: public/next.svg
Comment (Prabeshpd) [reply public/next.svg:1]: Updated the file

**Review by tednguyendev — APPROVED** [review]
LGTM!

You can also add a README file to this PR as well.

You may want to ref [this](https://github.com/nimblehq/js-framework-benchmark/blob/develop/nextjs-12/README.md).

File: src/app/globals.css
Comment (olivierobert) [inline src/app/globals.css:1]: Manifest files are never named `globals.css` but `application.css`. See https://github.com/nimblehq/js-framework-benchmark/blob/develop/nextjs-13/src/stylesheets/application.scss

File: src/app/globals.css [hunk 2]
```diff
@@ -0,0 +1,27 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+:root {
```
Comment (olivierobert) [inline src/app/globals.css:5]: `:root` should go into `base/_root.css`

File: src/app/globals.css [hunk 3]
```diff
@@ -0,0 +1,27 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
```
Comment (olivierobert) [inline src/app/globals.css:4]: As more style will be added it is better to start moving all the below into files that will be imported. You need to follow this directory structure: https://nimblehq.co/compass/development/code-conventions/css/#stylesheets-structure

File: src/app/globals.css [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@media (prefers-color-scheme: dark) {
+  :root {
+    --foreground-rgb: 255, 255, 255;
+    --background-start-rgb: 0, 0, 0;
+    --background-end-rgb: 0, 0, 0;
+  }
+}
+
+body {
```
Comment (olivierobert) [inline src/app/globals.css:19]: This should go into `layouts/default.css`

File: src/app/layout.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import './globals.css';
+import type { Metadata } from 'next';
+import { Inter } from 'next/font/google';
+
+const inter = Inter({ subsets: ['latin'] });
+
+export const metadata: Metadata = {
+  title: 'Create Next App',
+  description: 'Generated by create next app',
```
Comment (olivierobert) [inline src/app/layout.tsx:9]: You can already change this to the name or purpose of the application 😅

**Review by olivierobert — CHANGES_REQUESTED** [review]

File: src/app/layout.tsx (hunk 5 above)
Comment (Prabeshpd) [reply src/app/layout.tsx]: Fixed in [dd9e847](https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/25/commits/dd9e8478e9beffcb5721f700c122c1f6d1d84f24)

File: src/app/globals.css (hunk 4 above)
Comment (Prabeshpd) [reply src/app/globals.css]: Fixed in [4390bba](https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/25/commits/4390bba77bdc57ae645358f191396947b3490b80)

File: src/app/globals.css (hunk 3 above)
Comment (Prabeshpd) [reply src/app/globals.css]: Fixed in [4390bba](https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/25/commits/4390bba77bdc57ae645358f191396947b3490b8

File: src/app/globals.css
Comment (Prabeshpd) [reply src/app/globals.css:1]: Fixed in [4390bba](https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/25/commits/4390bba77bdc57ae645358f191396947b3490b8

**Review by olivierobert — APPROVED** [review]

**Pull request merged**